feat(hooks): enhance useRenderState with options object for rendering states

yoonyounghoon opened the pull request:
Added declarative render options to make `RenderState` usage more intuitive.

- Added `RenderOptions` interface to support onIdle, onLoading, onSuccess, and onError callbacks.
- Updated `render` function to accept either individual render functions or a single options object.
- Expanded test cases to cover state transitions using the new options object.

stegano commented on src/hooks/use-render-state.interface.ts:
```diff
@@ -76,15 +76,30 @@ export interface RenderSuccess<Data, Error> {
   (renderSuccess?: RenderSuccess<Data, Error>): ReactNode;
+  (
+    renderSuccess?: RenderSuccess<Data, Error>,
+    renderLoading?: RenderLoading<Data, Error>,
```
I think we only need :98–:103, since :94–:97 are already covered within those lines.

stegano commented on src/hooks/use-render-state.interface.ts:
```diff
@@ -76,15 +76,30 @@ export interface RenderSuccess<Data, Error> {
+ * RenderOptions
+ */
+export interface RenderOptions<Data, Error> {
+  onIdle?: RenderIdle<Data, Error>;
+  onLoading?: RenderLoading<Data, Error>;
+  onSuccess?: RenderSuccess<Data, Error>;
+  onError?: RenderError<Data, Error>;
```
I suggest using the prefix `render~` for render functions, because it makes the code more consistent.

yoonyounghoon commented:
@stegano Thanks for the review! Updated the code according to the comments.

- change interface key prefix from 'on' to 'render'
- remove redundant type definition
- Updated test cases according to the prefix changes

stegano commented:
@all-contributors please add @yoonyounghoon for code

stegano commented:
@all-contributors please add @yoonyounghoon for code and idea

stegano commented:
Thanks for your PR! This feature has been released in [v2.1.3](https://www.npmjs.com/package/react-render-state)
